### PR TITLE
Add directional variants to APC sprites

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do

This PR does several things with APCs.

- It creates rotational versions of all APC sprites.
- It ports the tgstation macro for creating directional variants of objects for ease of use in mapping.
- It creates an exhaustive UpdatePaths script for every single APC.
- It runs the UpdatePaths script on every map.
- It splits APCs varedited with cell charges to "important" and "critical" APCs. Important APCs were either 5,000 or 15,000 and were brought up to 15,000. Critical APCs were 25,000 and kept that value.
- It removes names for nearly every APC in the cases where their name matched their area or their purpose was subsumed into their subtype. An APC with an area autoname and a subtype indicating it is "important" is more useful than an APC called "Cryo and Arrivals" with a varedited cell_charge.

The icon states were generated using [ss13_make_rotationals][], with the east/west and south/north directions flipped. I'd much prefer we have sprites where the south-facing sprite is the one used for south bumps, rather then the south-facing sprite being used for north bumps, which just introduces confusion.

[ss13_make_rotationals]: https://github.com/warriorstar-orion/ss13_make_rotationals

<details><summary><i>Rant about UpdatePaths</i></summary>
For the record, I do not like UpdatePath's syntax. When I write automation to generate UpdatePaths scripts, they are typically completely explicit. There is no use of `@OLD`, `@SKIP`, etc. because not only is automation that is exhaustive easier to write, but scripts that use those shortcuts are difficult to reason about, harder to verify, and fail under even slightly complex migrations. For example, I need to migrate `/obj/machinery/power/apc/off_station` and `/obj/machinery/power/apc/off_station/empty_charge`. You'd think `@SUBTYPES` is where this comes in. But no, because the change I'm making is appending to the existing path, and so `/obj/machinery/power/apc/off_station/@SUBTYPES/directional/north` doesn't work. The other problem with this is that if you have reflection available to you, and all the paths are explicit, you can actually check to see if, you know, the paths you're creating actually exist in the codebase. And, ultimately, every UpdatePaths migration is going to have some edge cases that need to be explicit, so you might as well make the whole script explicit. I also think var replacements like <code>{@OLD;cell_type=@SKIP;dir=@SKIP;keep_preset_name=@SKIP;name=@SKIP;pixel_y=@SKIP;req_access=@SKIP}</code> are _completely_ unreadable.

But I also know that if I did have a more explicit script, I'd get pushback from people asking why I didn't use the `@` helpers. So I'm just heading it off at the pass.

You can see an explicit version of the script [here][]. It is easy to read, easy to understand exactly what varedits will be changed, makes it obvious what varedits we do care about keeping (such as `locked`, `equipment_channel` and so on), and is much more useful as documentation of the state of the code at the time it was written. Is it as future-proof? Probably not. But decisions like this are a question of trade-offs, and I think the trade-offs are worth it.

[here]: https://gist.githubusercontent.com/warriorstar-orion/6e9d9d17f723c0158b000064687dfecd/raw/aa481abe4541e9d5b8fc9884c1d9c1ac99bf8669/21455_apc_directionals.txt

</details>

## Why It's Good For The Game

We have far too many APCs on maps with arbitrary and at some times flatly wrong varedits. This matches up every single APC on all our maps with a specific subtype, with only a handful of varedits needed. This should also bring us in line with how other forks map APCs, which should prevent any hiccups when rendering APCs in MDB.

### Before

2024_02_12__14_55_08__Paradise Station 13
2024_02_12__14_58_29__paradise dme  cyberiad dmm  - StrongDMM

### After

2024_02_12__13_58_38__paradise dme  cyberiad dmm  - StrongDMM

### Building an APC

2024_02_12__14_33_51__Paradise Station 13
2024_02_12__14_34_40__Paradise Station 13
2024_02_12__14_34_43__Paradise Station 13
2024_02_12__14_34_48__Paradise Station 13

## Testing

Spot-checked maps in SDMM and in-game, verified building an APC sets direction properly.

## Changelog

:cl:
fix: APCs are rotated based on their wall position.
/:cl:

### CODE REVIEW

**Comment by warriorstar-orion** [comment]
For reference, these two sprites appear in IDB:
2024_02_12__15_28_34__Add directional variants to APC sprites by warriorstar-orion · Pull Request #241
2024_02_12__15_28_38__Add directional variants to APC sprites by warriorstar-orion · Pull Request #241

Because they are non-animated sprites which had their rewind bit set, which the DMI rust library (which I used to make some of this PR) automatically keeps unset if an icon only has one frame:

https://github.com/spacestation13/dmi-rust/blob/1861df71bfb2eb70ca2b0379f8746c89636c95b3/src/icon.rs#L324-L344

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 1]
```diff
@@ -999,9 +1000,9 @@
 	},
 /obj/machinery/power/apc{
 	cell_type = 5000;
-	dir = 8;
-	name = "west bump Important Area";
```
Comment (warriorstar-orion) [inline _maps/map_files/cyberiad/cyberiad.dmm:1003]: I know I wiped the name of "important" APCs like this one, in another pass I'm going to make an important APC subtype if we don't already have one

**Comment by warriorstar-orion** [comment]
Wait no I fucked it
2024_02_12__16_31_50__0-after png (6464×5152)

File: code/__DEFINES/mappers.dm [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+##path/directional/east {\
+	name = "east bump"; \
+	dir = EAST; \
+	pixel_x = offset; \
+} \
+##path/directional/west {\
+	name = "west bump"; \
+	dir = WEST; \
+	pixel_x = -offset; \
+}
```
Comment (S34NW) [inline code/__DEFINES/mappers.dm:21]:
```suggestion
#define MAPPING_DIRECTIONAL_HELPERS(path, offset_y, offset_x) ##path/directional/north {\
	name = "north bump"; \
	dir = NORTH; \
	pixel_y = offset_y; \
} \
##path/directional/south {\
	name = "south bump"; \
	dir = SOUTH; \
	pixel_y = -offset_y; \
} \
##path/directional/east {\
	name = "east bump"; \
	dir = EAST; \
	pixel_x = offset_x; \
} \
##path/directional/west {\
	name = "west bump"; \
	dir = WEST; \
	pixel_x = -offset_x; \
}
```

Would be better to allow split x/y offsets like this?

Comment (warriorstar-orion) [reply code/__DEFINES/mappers.dm]: SGTM

**Review by S34NW — APPROVED** [review]

**Pull request merged**